### PR TITLE
Refactor test_unavailability

### LINKED ISSUE #2112: Timeout in test_wal_acceptor::test_unavailability
https://app.circleci.com/pipelines/github/neondatabase/neon/8150/workflows/ef2c3e04-1585-40c3-802b-7ebe7611c673/jobs/83441/steps

```
=================================== FAILURES ===================================
_____________________________ test_unavailability ______________________________
[gw2] linux -- Python 3.9.2 /home/runner/.cache/pypoetry/virtualenvs/zenith-TcEEkJmQ-py3.9/bin/python
test_runner/batch_others/test_wal_acceptor.py:253: in test_unavailability
    proc.join()
../.pyenv/versions/3.9.2/lib/python3.9/multiprocessing/process.py:149: in join
    res = self._popen.wait(timeout)
../.pyenv/versions/3.9.2/lib/python3.9/multiprocessing/popen_fork.py:43: in wait
    return self.poll(os.WNOHANG if timeout == 0.0 else 0)
../.pyenv/versions/3.9.2/lib/python3.9/multiprocessing/popen_fork.py:27: in poll
    pid, sts = os.waitpid(self.pid, flag)
E   Failed: Timeout >300.0s
------------------------------ Captured log setup ------------------------------
2022-07-18 10:48:57.266 INFO [neon_fixtures.py:2047] get_test_output_dir is /tmp/test_output/test_unavailability
2022-07-18 10:48:57.266 INFO [neon_fixtures.py:2068] test_output_dir is /tmp/test_output/test_unavailability
----------------------------- Captured stderr call -----------------------------
```

### PR DESCRIPTION
**Pull request by petuhovskiy**
Can fix #2112, or at least make debugging easier

### CODE REVIEW

File: test_runner/batch_others/test_wal_acceptor_async.py [hunk 1]
```diff
@@ -404,3 +404,55 @@ def test_concurrent_computes(neon_env_builder: NeonEnvBuilder):
...
+                               key: int,
+                               start_delay_sec: int = 2):
+    # shutdown one of two acceptors, that is, majority
+    sk.stop()
+
+    bg_query = asyncio.create_task(conn.execute(f"INSERT INTO t values ({key}, 'payload')"))
+
+    await asyncio.sleep(start_delay_sec)
+    # ensure that the query has not been executed yet
+    assert not bg_query.done()
```
Comment (hlinnaka) [inline test_runner/batch_others/test_wal_acceptor_async.py:422]: I'm not very familiar with python async: is the task ever "done", if you don't call wait on it?

File: test_runner/batch_others/test_wal_acceptor_async.py [hunk 2]
```diff
@@ -404,3 +404,50 @@ def test_concurrent_computes(neon_env_builder: NeonEnvBuilder):
...
+
+    # check basic work with table
+    await conn.execute('CREATE TABLE t(key int primary key, value text)')
+    await conn.execute("INSERT INTO t values (1, 'payload')")
+
+    # shutdown one of two acceptors, that is, majority
+    env.safekeepers[0].stop()
+
+    delayed_start = asyncio.create_task(delayed_safekeeper_start(env.safekeepers[0], start_delay_sec))
+    start = time.time()
```
[the remark this answers is not in the document]
Comment (petuhovskiy) [reply test_runner/batch_others/test_wal_acceptor_async.py]: I guess this race condition was in the test from the creation. Haven't seen a failure in practice, but better to be more robust. Refactored as in the 2nd proposal.

File: test_runner/batch_others/test_wal_acceptor_async.py (hunk 1 above)
Comment (petuhovskiy) [reply test_runner/batch_others/test_wal_acceptor_async.py:422]: I'm also not an expert, but `create_task` should start it in the background.

Comment (hlinnaka) [reply test_runner/batch_others/test_wal_acceptor_async.py:422]: You could comment out the `sk.stop()` above, and verify that the assertion fails. Or just use `wait` to wait for both tasks concurrently, to be safe.

Comment (petuhovskiy) [reply test_runner/batch_others/test_wal_acceptor_async.py:422]: https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task

> Wrap the coro [coroutine](https://docs.python.org/3/library/asyncio-task.html#coroutine) into a [Task](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task) and schedule its execution. Return the Task object.

Comment (petuhovskiy) [reply test_runner/batch_others/test_wal_acceptor_async.py:422]: > You could comment out the sk.stop() above, and verify that the assertion fails.

It fails, as expected :)

**Review by hlinnaka — APPROVED** [review]

**Pull request merged**